Cryptonight rework

UselessGuru opened the pull request:
**Please test - I cannot test myself at the moment!!!**

This PR enables all 4 cryptonight algorithms

This whole monero / Cryptonight fork is a mess! See here:
https://github.com/xmrig/xmrig/issues/482

There are now 4 different cryptonight algorithms:
Cryptonight (old)
Cryptonight-Lite
Cryptonight-Heavy
CryptonightV7
And it all depends on the coin name. Grr...

Only two MPM supported pools allow mining cryptonight: Nicehash & MPH/MPHC

Currently only NiceHash properly list CryptonightV7 as new algorithm.
Luckily MPH / MPHC exposes the coin name in the API. As a workaround for MPH / MPHC I've added a function 'Get-CryptonightAlgorithm' which returns the proper Cryptonight algorithm based on coin name.

symalla commented:
Still testing  although I wish it was correct it seems the profit for MiningpoolHubCoins-Monero might be a little off.

![2018-04-11 11_48_54-192 168 1 33 - remote desktop connection](https://user-images.githubusercontent.com/7012032/38631167-a13e4bb6-3d7e-11e8-900e-cfdbfbaba50b.jpg)

unikon2450 commented:
Hello. I changed the files but something is not working properly. Can you check what I did wrong?

[MultiPoolMiner_2018-04-12_11-02-17.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1902356/MultiPoolMiner_2018-04-12_11-02-17.txt)

station384 commented:
The change to cryptonightv7 seems to be working, (picking correctly on nicehash) 

also getting the same error as reported by @unikon2450

UselessGuru commented:
@station384 & @unikon2450 

Thank you for your valuable feedback!

I've made some changes which hopefully fix the reported issue. Please try again and report back. TIA!

unikon2450 commented:
I still have a problem with MPH. MPM does not work on this pool. MPM does not even begin to test miners in cryptonight.
I also used the default start file. Cryptonight is benchmarked only on nicehash.
[MPH start.bat MultiPoolMiner_2018-04-12_18-05-17.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1904180/MPH.start.bat.MultiPoolMiner_2018-04-12_18-05-17.txt)
[start.bat MultiPoolMiner_2018-04-12_18-14-18.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1904181/start.bat.MultiPoolMiner_2018-04-12_18-14-18.txt)

station384 commented:
@UselessGuru So far I have not been getting the error that it can't find file that @unikon2450 mentioned when mining nicehash,  I have all the tests of Claymore going, and they have all passed and appear to work correctly.


I can confirm @unikon2450 that MPM is not even testing benchmarks for cryptonight on MultiPoolHub.  It appears it isn't matching that it is an option on that pool.

grantemsley commented:
Thanks @station384 - looks like you are correct, I've made those changes and the algorithms show up now.

Seeing a lot of failed connection issues on the miners though - not sure if pool issues or it's not setting the pool parameters properly.

symalla commented:
It looks like the miners are trying to use the URL of us-east.hub.miningpoolhub.com:20580 instead of us-east.cryptonight-hub.miningpoolhub.com:20580

UselessGuru commented:
@symalla

MPH / MPHC send wrong data through it API...
API says: 'host: hub.miningpoolhub.com'
Web page says: [location].cryptonight-hub.miningpoolhub.com

I'll fix the pool files.

n3cr0cr0w commented:
US should be us-east.cryptonight-hub.miningpoolhub.com:20580 and not us.cryptonight-hub.miningpoolhub.com.

https://monero.miningpoolhub.com/

UselessGuru commented:
Yet another try fixing the cryptonight mess :)

`currency = "monero7"`

Fixed. For Fireice (Xmr-Stak) Currency will be set if pool sends it by API.

`Host = "us-east.cryptonight-$($MiningPoolHub_Hosts`

Fixed. Changed MPH / MPHC url for cryptonight

I am interested how (if???) cryptonight mining is working on nicehash - nicehash lists cryptonight AND cryptonight7 as available algorithms, but does not say what currency they are mining with. 
??? How should that work If Xmr-Stak requires the correct currency to be specified ???

angelics commented:
based on algorithm.txt and past ccminer "algo" , (eg. poly, polytimos)
isnt it much "proper" way to have monero7 in algorithm.txt rather than having it to convert inside miner.ps1?

aaronsta1 commented:
seems to work but after awhile pwsh.exe takes 100% system memory.. didnt do it on the main branch.

erwan-joly commented:
this doesn't seems to work. My vega 64 are a lot better for cryptonight v7 and the benchmark select ETH+pasc

angelics commented:
@UselessGuru
[FireiceNvidia242.zip](https://github.com/MultiPoolMiner/MultiPoolMiner/files/1953862/FireiceNvidia242.zip)

i added algo into algorigthm.txt. and do not have include.ps1 as urs here.
last time i checked it works,as an nvidia user i don particularly mine xmr or xmr pool. 
since there is no recommended pools (mph,zpool) support any other coin than monerov7, shudnt we just leave other coins out atm? pool operators and miner dev messed up with the names atm.

grantemsley commented:
I'm testing this.  Not done yet, but have some questions/comments already:

- Is miningpoolhub the only pool we have that supports cryptonight<whatever> algorithms?
- MiningPoolHubCoins API reports 2 coins that use Cryptonight-Monero (their name for CryptonightV7): Monero and Electroneum.  But isn't Electroneum still on the regular old Cryptonight?  Everything I can find says it is.  Is their API wrong?  Is that pool even working???
- I'd recommend ditching the Get-CryptonightAlgorithm function and do the conversions from pool name to real algorithm name in the pool files.  Each pool seems to be handling it differently and its probably safer to tackle each pool one by one.
- Between the pools and the miner authors nobody came up with a single naming convention for this stuff?  Why?  What is wrong with people!

mkirov0 commented:
KlausT has a Cryptonight only miner out. Posting just FYI.
https://github.com/KlausT/ccminer-cryptonight
